src/test/java/teammates/test/cases/webapi/GetStudentProfileActionTest.java

```diff
@@ -28,80 +30,232 @@ protected String getRequestMethod() {
     public void testExecute() throws Exception {
-        AccountAttributes student1 = typicalBundle.accounts.get("student1InCourse1");
+        // See test cases below.
+    }
 
-        StudentProfileAttributes student1InCourse1Profile = typicalBundle.profiles.get("student1InCourse1");
-        testActionSuccess(student1, student1InCourse1Profile, "Typical case");
+    @Test
+    public void testExecute_withExistingProfileAndNoParameter_shouldReturnOwnProfile() {
+        AccountAttributes student1InCourse1 = typicalBundle.accounts.get("student1InCourse1");
+        StudentProfileAttributes expectedProfile = typicalBundle.profiles.get("student1InCourse1");
+        String expectedName = student1InCourse1.name;
+        loginAsStudent(student1InCourse1.googleId);
+        testGetCorrectProfile(expectedProfile, expectedName);
+    }
 
-        testActionInMasquerade(student1);
+    @Test
+    public void testExecute_withMissingCourseId_shouldReturnOwnProfile() {
+        AccountAttributes student1InCourse1 = typicalBundle.accounts.get("student1InCourse1");
+        StudentProfileAttributes expectedProfile = typicalBundle.profiles.get("student1InCourse1");
+        String expectedName = student1InCourse1.name;
+        loginAsStudent(student1InCourse1.googleId);
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.STUDENT_EMAIL, student1InCourse1.getEmail(),
+        };
+        testGetCorrectProfile(expectedProfile, expectedName, submissionParams);
+    }
 
-        student1 = typicalBundle.accounts.get("student1InTestingSanitizationCourse");
-        StudentProfileAttributes student1InTestingSanitizationCourseProfile =
-                typicalBundle.profiles.get("student1InTestingSanitizationCourse");
-        // simulate sanitization that occurs before persistence
-        student1.sanitizeForSaving();
-        testActionSuccess(student1, student1InTestingSanitizationCourseProfile,
-                "Typical case: attempted script injection");
+    @Test
+    public void testExecute_withMissingStudentEmail_shouldReturnOwnProfile() {
+        StudentAttributes student1InCourse1 = typicalBundle.students.get("student1InCourse1");
+        StudentProfileAttributes expectedProfile = typicalBundle.profiles.get("student1InCourse1");
+        String expectedName = typicalBundle.accounts.get("student1InCourse1").name;
+        loginAsStudent(student1InCourse1.googleId);
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, student1InCourse1.getCourse(),
+        };
+        testGetCorrectProfile(expectedProfile, expectedName, submissionParams);
     }
 
-    private void testActionSuccess(AccountAttributes student, StudentProfileAttributes studentProfileAttributes,
-                                   String caseDescription) {
-        loginAsStudent(student.googleId);
+    @Test
+    public void testExecute_withStudentEmailAndCourseId_shouldReturnProfileByStudentEmail() {
```

Comment:
I think it is necessary to specify the request entity in the naming. Same in other places?

Reply:
I think it's unnecessary, because the logic of `execute()` is blind to who request the profile (i.e. the request entity), The entity is only important in `checkAccessControl`